include/lsst/afw/fits.h

```diff
@@ -631,6 +631,29 @@ class Fits {
     int behavior;  // bitwise OR of BehaviorFlags
 };
 
+/**
+ * Combine two sets of metadata in a FITS-safe fashion
+ *
+ * If name is "COMMENT" or "HISTORY" then:
+ * - The item is valid if it of type std::string (whether or not it is an array)
+ * - Valid metadata from `second` is appended to same-named valid metadata in `first`.
+ * Otherwise:
+ * - The item is valid if it is a scalar (regardless of type)
+ * - Valid metadata from `second` supersedes same-named metadata from `first`
+ *   (even if the types are different). Name order for matching items
+ *   should not be assumed, since it relies on internal details that may change.
```

Comment:
I think it ought to be well-defined: the ordering from `first` is used with any new items from `second` added afterwards.